ci-operator/step-registry/ibmcloud/provision/dedicatedhost/ibmcloud-provision-dedicatedhost-commands.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#Since sometimes fail to create the dedicated host in some zone, so just continue test when create dedicated host succeed, and create cluster just on one zone. 
+#Known Issues: 
+# OCPBUGS-5906 [IPI-IBMCloud] the worker nodes are “Provisioning” when using the dedicated host in the Default group
+# OCPBUGS-18925 [IPI-IBMCloud] ] fail to use vx2d-host-176x2464 profile with dedicated host in install-config
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+
+function create_resource_group() {
+    local rg="$1"
+    echo "create resource group ... ${rg}"
+    "${IBMCLOUD_CLI}" resource group-create ${rg} || return 1
+    echo "${rg}" > "${SHARED_DIR}/ibmcloud_resource_group"
+    "${IBMCLOUD_CLI}" target -g ${rg} || return 1
+}
+
+#random select the group which create the dedicatd host
+function randomSelectRG {
+    dhGroups=("Default" "${resource_group}")
+    selected_idx=$((RANDOM % 2))
+    rg=${dhGroups[selected_idx]}
+    echo $rg
+}
+
+function randomGetDHProfile {
+    local all_profiles profiles selected_id proTotal
+    all_profiles=$(${IBMCLOUD_CLI} is dhps --output JSON | jq -r '.[] | select(.family != "very-high-memory") | .name')
+    mapfile -t profiles <<< "${all_profiles}"
+    proTotal=${#profiles[@]}
+    selected_id=$((RANDOM % ${proTotal}))
+    echo ${profiles[$selected_id]}
+}
+
+function createDH() {
+    local rg=$1 dProfile=$2 zone=$3 dhName=$4
+    local tmp family class dhg status
+    if [[ ${rg} = "Default" ]]; then
+        run_command "${IBMCLOUD_CLI} target --unset-resource-group"
+    else
+        run_command "${IBMCLOUD_CLI} target -g $rg"
+    fi
+    run_command "${IBMCLOUD_CLI} is target --gen 2"
+    tmp=$(${IBMCLOUD_CLI} is dedicated-host-profile $dProfile --output JSON | jq -r '.family +","+ .class')
+    family=${tmp%,*}
+    class=${tmp#*,}
+    echo "family:" $family " class:" $class
+    dhg="${dhName}g"
+    dhg=$(${IBMCLOUD_CLI} is dedicated-host-group-create --zone $zone --family $family --class $class --name ${dhg} --output JSON | jq -r .id)
+    
+    ${IBMCLOUD_CLI} is dhg $dhg
+
+    ${IBMCLOUD_CLI} is dedicated-host-create --profile $dProfile --name $dhName --dhg $dhg --output JSON
```

Comment:
I saw `--dhg` support name of the host group, not have to be ID, to help code reading, suggest to directly use dhg name, remove the dhg ID var.